sqlflow client doesn't build for Windows

[Building](https://travis-ci.com/github/sql-machine-learning/sqlflow/jobs/335373529#L284) in the Travis CI Windows VM reports the [error](https://travis-ci.com/github/sql-machine-learning/sqlflow/jobs/335373529#L338):

```
# sqlflow.org/sqlflow/cmd/sqlflow
cmd\sqlflow\main.go:352:79: cannot use syscall.Stdin (type syscall.Handle) as type int in argument to terminal.IsTerminal
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -105,6 +105,7 @@ jobs:
     - $TRAVIS_BUILD_DIR/scripts/travis/deploy_client.sh
   - env: DESC="Deploy Linux client"
     os: linux
+    dist: bionic
     script:
     - $TRAVIS_BUILD_DIR/scripts/travis/deploy_client.sh
   - env: DESC="Deploy Windows client"

--- a/.travis.yml
+++ b/.travis.yml
@@ -34,70 +34,70 @@ install:
 - echo "skip install"
 jobs:
   include:
-  # - stage: BuildAndTest
-  #   script:
-  #   - set -e
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-  #   - docker run --rm -it -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:dev
-  #     pre-commit run -a --show-diff-on-failure
-  # - env: SQLFLOW_TEST_DB=mysql
-  #   script:
-  #   - set -e
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-  #   - docker run --rm -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:ci
-  #     scripts/test/units.sh
-  #   - docker run --rm -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:ci
-  #     scripts/upload_coveralls.sh
-  # - env: SQLFLOW_TEST_DB=hive # run more parallel tests in the same stage:
-  #   script:
-  #   - set -e
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-  #   - docker pull sqlflow/gohive:dev
-  #   - docker run -d --name=hive sqlflow/gohive:dev python3 -m http.server 8899
-  #   - docker run --rm -it --net=container:hive
-  #     -v $TRAVIS_BUILD_DIR:/work -w /work
-  #     -v $GOPATH:/root/go
-  #     sqlflow:ci scripts/test/hive.sh
-  # - env: SQLFLOW_TEST_DB=maxcompute
-  #   script:
-  #   - set -e
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/setup_minikube.sh
-  #   - docker run --rm --net=host
-  #     -v /var/run/docker.sock:/var/run/docker.sock
-  #     -v $HOME/.kube:/root/.kube
-  #     -v /home/$USER/.minikube/:/home/$USER/.minikube/
-  #     -v $TRAVIS_BUILD_DIR:/work -w /work
-  #     sqlflow:ci scripts/test/ipython.sh
-  #   - docker run --rm --net=host
-  #     -e SQLFLOW_TEST_DB_MAXCOMPUTE_AK=$MAXCOMPUTE_AK
-  #     -e SQLFLOW_TEST_DB_MAXCOMPUTE_SK=$MAXCOMPUTE_SK
-  #     -v /var/run/docker.sock:/var/run/docker.sock
-  #     -v $HOME/.kube:/root/.kube
-  #     -v /home/$USER/.minikube/:/home/$USER/.minikube/
-  #     -v $TRAVIS_BUILD_DIR:/work -w /work
-  #     sqlflow:ci scripts/test/maxcompute.sh
-  # - env: SQLFLOW_TEST=java
-  #   script:
-  #   - set -e
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-  #   - docker run --rm
-  #     -v $TRAVIS_BUILD_DIR:/work -w /work
-  #     sqlflow:ci scripts/test/java.sh
-  # - env: SQLFLOW_TEST=workflow
-  #   script:
-  #   - set -e
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/setup_minikube.sh
-  #   - eval $(sudo minikube docker-env)
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-  #   - docker tag sqlflow:ci sqlflow:submitter
-  #   - docker run --rm --net=host
-  #     -e SQLFLOW_WORKFLOW_STEP_IMAGE=sqlflow:submitter
-  #     -v /var/run/docker.sock:/var/run/docker.sock
-  #     -v $HOME/.kube:/root/.kube
-  #     -v /home/$USER/.minikube/:/home/$USER/.minikube/
-  #     -v $TRAVIS_BUILD_DIR:/work -w /work
-  #     sqlflow:ci scripts/test/workflow.sh
+  - stage: BuildAndTest
+    script:
+    - set -e
+    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+    - docker run --rm -it -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:dev
+      pre-commit run -a --show-diff-on-failure
+  - env: SQLFLOW_TEST_DB=mysql
+    script:
+    - set -e
+    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+    - docker run --rm -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:ci
+      scripts/test/units.sh
+    - docker run --rm -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:ci
+      scripts/upload_coveralls.sh
+  - env: SQLFLOW_TEST_DB=hive # run more parallel tests in the same stage:
+    script:
+    - set -e
+    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+    - docker pull sqlflow/gohive:dev
+    - docker run -d --name=hive sqlflow/gohive:dev python3 -m http.server 8899
+    - docker run --rm -it --net=container:hive
+      -v $TRAVIS_BUILD_DIR:/work -w /work
+      -v $GOPATH:/root/go
+      sqlflow:ci scripts/test/hive.sh
+  - env: SQLFLOW_TEST_DB=maxcompute
+    script:
+    - set -e
+    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+    - $TRAVIS_BUILD_DIR/scripts/travis/setup_minikube.sh
+    - docker run --rm --net=host
+      -v /var/run/docker.sock:/var/run/docker.sock
+      -v $HOME/.kube:/root/.kube
+      -v /home/$USER/.minikube/:/home/$USER/.minikube/
+      -v $TRAVIS_BUILD_DIR:/work -w /work
+      sqlflow:ci scripts/test/ipython.sh
+    - docker run --rm --net=host
+      -e SQLFLOW_TEST_DB_MAXCOMPUTE_AK=$MAXCOMPUTE_AK
+      -e SQLFLOW_TEST_DB_MAXCOMPUTE_SK=$MAXCOMPUTE_SK
+      -v /var/run/docker.sock:/var/run/docker.sock
+      -v $HOME/.kube:/root/.kube
+      -v /home/$USER/.minikube/:/home/$USER/.minikube/
+      -v $TRAVIS_BUILD_DIR:/work -w /work
+      sqlflow:ci scripts/test/maxcompute.sh
+  - env: SQLFLOW_TEST=java
+    script:
+    - set -e
+    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+    - docker run --rm
+      -v $TRAVIS_BUILD_DIR:/work -w /work
+      sqlflow:ci scripts/test/java.sh
+  - env: SQLFLOW_TEST=workflow
+    script:
+    - set -e
+    - $TRAVIS_BUILD_DIR/scripts/travis/setup_minikube.sh
+    - eval $(sudo minikube docker-env)
+    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+    - docker tag sqlflow:ci sqlflow:submitter
+    - docker run --rm --net=host
+      -e SQLFLOW_WORKFLOW_STEP_IMAGE=sqlflow:submitter
+      -v /var/run/docker.sock:/var/run/docker.sock
+      -v $HOME/.kube:/root/.kube
+      -v /home/$USER/.minikube/:/home/$USER/.minikube/
+      -v $TRAVIS_BUILD_DIR:/work -w /work
+      sqlflow:ci scripts/test/workflow.sh
   - stage: Deploy
     env: DESC="Deploy macOS client"
     os: osx
@@ -112,6 +112,6 @@ jobs:
     os: windows
     script:
     - $TRAVIS_BUILD_DIR/scripts/travis/deploy_client.sh
-  # - env: DESC="Deploy server Docker image"
-  #   script:
-  #   - $TRAVIS_BUILD_DIR/scripts/travis/deploy_server.sh
+  - env: DESC="Deploy server Docker image"
+    script:
+    - $TRAVIS_BUILD_DIR/scripts/travis/deploy_server.sh

--- a/.travis.yml
+++ b/.travis.yml
@@ -34,75 +34,72 @@ install:
 - echo "skip install"
 jobs:
   include:
-  - stage: BuildAndTest
-    script:
-    - set -e
-    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-    - docker run --rm -it -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:dev
-      pre-commit run -a --show-diff-on-failure
-  - env: SQLFLOW_TEST_DB=mysql
-    script:
-    - set -e
-    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-    - docker run --rm -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:ci
-      scripts/test/units.sh
-    - docker run --rm -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:ci
-      scripts/upload_coveralls.sh
-  - env: SQLFLOW_TEST_DB=hive # run more parallel tests in the same stage:
-    script:
-    - set -e
-    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-    - docker pull sqlflow/gohive:dev
-    - docker run -d --name=hive sqlflow/gohive:dev python3 -m http.server 8899
-    - docker run --rm -it --net=container:hive
-      -v $TRAVIS_BUILD_DIR:/work -w /work
-      -v $GOPATH:/root/go
-      sqlflow:ci scripts/test/hive.sh
-  - env: SQLFLOW_TEST_DB=maxcompute
-    script:
-    - set -e
-    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-    - $TRAVIS_BUILD_DIR/scripts/travis/setup_minikube.sh
-    - docker run --rm --net=host
-      -v /var/run/docker.sock:/var/run/docker.sock
-      -v $HOME/.kube:/root/.kube
-      -v /home/$USER/.minikube/:/home/$USER/.minikube/
-      -v $TRAVIS_BUILD_DIR:/work -w /work
-      sqlflow:ci scripts/test/ipython.sh
-    - docker run --rm --net=host
-      -e SQLFLOW_TEST_DB_MAXCOMPUTE_AK=$MAXCOMPUTE_AK
-      -e SQLFLOW_TEST_DB_MAXCOMPUTE_SK=$MAXCOMPUTE_SK
-      -v /var/run/docker.sock:/var/run/docker.sock
-      -v $HOME/.kube:/root/.kube
-      -v /home/$USER/.minikube/:/home/$USER/.minikube/
-      -v $TRAVIS_BUILD_DIR:/work -w /work
-      sqlflow:ci scripts/test/maxcompute.sh
-  - env: SQLFLOW_TEST=java
-    script:
-    - set -e
-    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-    - docker run --rm
-      -v $TRAVIS_BUILD_DIR:/work -w /work
-      sqlflow:ci scripts/test/java.sh
-  - env: SQLFLOW_TEST=workflow
-    script:
-    - set -e
-    - $TRAVIS_BUILD_DIR/scripts/travis/setup_minikube.sh
-    - eval $(sudo minikube docker-env)
-    - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
-    - docker tag sqlflow:ci sqlflow:submitter
-    - docker run --rm --net=host
-      -e SQLFLOW_WORKFLOW_STEP_IMAGE=sqlflow:submitter
-      -v /var/run/docker.sock:/var/run/docker.sock
-      -v $HOME/.kube:/root/.kube
-      -v /home/$USER/.minikube/:/home/$USER/.minikube/
-      -v $TRAVIS_BUILD_DIR:/work -w /work
-      sqlflow:ci scripts/test/workflow.sh
+  # - stage: BuildAndTest
+  #   script:
+  #   - set -e
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+  #   - docker run --rm -it -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:dev
+  #     pre-commit run -a --show-diff-on-failure
+  # - env: SQLFLOW_TEST_DB=mysql
+  #   script:
+  #   - set -e
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+  #   - docker run --rm -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:ci
+  #     scripts/test/units.sh
+  #   - docker run --rm -v $TRAVIS_BUILD_DIR:/work -w /work sqlflow:ci
+  #     scripts/upload_coveralls.sh
+  # - env: SQLFLOW_TEST_DB=hive # run more parallel tests in the same stage:
+  #   script:
+  #   - set -e
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+  #   - docker pull sqlflow/gohive:dev
+  #   - docker run -d --name=hive sqlflow/gohive:dev python3 -m http.server 8899
+  #   - docker run --rm -it --net=container:hive
+  #     -v $TRAVIS_BUILD_DIR:/work -w /work
+  #     -v $GOPATH:/root/go
+  #     sqlflow:ci scripts/test/hive.sh
+  # - env: SQLFLOW_TEST_DB=maxcompute
+  #   script:
+  #   - set -e
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/setup_minikube.sh
+  #   - docker run --rm --net=host
+  #     -v /var/run/docker.sock:/var/run/docker.sock
+  #     -v $HOME/.kube:/root/.kube
+  #     -v /home/$USER/.minikube/:/home/$USER/.minikube/
+  #     -v $TRAVIS_BUILD_DIR:/work -w /work
+  #     sqlflow:ci scripts/test/ipython.sh
+  #   - docker run --rm --net=host
+  #     -e SQLFLOW_TEST_DB_MAXCOMPUTE_AK=$MAXCOMPUTE_AK
+  #     -e SQLFLOW_TEST_DB_MAXCOMPUTE_SK=$MAXCOMPUTE_SK
+  #     -v /var/run/docker.sock:/var/run/docker.sock
+  #     -v $HOME/.kube:/root/.kube
+  #     -v /home/$USER/.minikube/:/home/$USER/.minikube/
+  #     -v $TRAVIS_BUILD_DIR:/work -w /work
+  #     sqlflow:ci scripts/test/maxcompute.sh
+  # - env: SQLFLOW_TEST=java
+  #   script:
+  #   - set -e
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+  #   - docker run --rm
+  #     -v $TRAVIS_BUILD_DIR:/work -w /work
+  #     sqlflow:ci scripts/test/java.sh
+  # - env: SQLFLOW_TEST=workflow
+  #   script:
+  #   - set -e
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/setup_minikube.sh
+  #   - eval $(sudo minikube docker-env)
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/build.sh
+  #   - docker tag sqlflow:ci sqlflow:submitter
+  #   - docker run --rm --net=host
+  #     -e SQLFLOW_WORKFLOW_STEP_IMAGE=sqlflow:submitter
+  #     -v /var/run/docker.sock:/var/run/docker.sock
+  #     -v $HOME/.kube:/root/.kube
+  #     -v /home/$USER/.minikube/:/home/$USER/.minikube/
+  #     -v $TRAVIS_BUILD_DIR:/work -w /work
+  #     sqlflow:ci scripts/test/workflow.sh
   - stage: Deploy
-    env: DESC="Deploy server Docker image"
-    script:
-    - $TRAVIS_BUILD_DIR/scripts/travis/deploy_server.sh
-  - env: DESC="Deploy macOS client"
+    env: DESC="Deploy macOS client"
     os: osx
     script:
     - $TRAVIS_BUILD_DIR/scripts/travis/deploy_client.sh
@@ -114,3 +111,6 @@ jobs:
     os: windows
     script:
     - $TRAVIS_BUILD_DIR/scripts/travis/deploy_client.sh
+  # - env: DESC="Deploy server Docker image"
+  #   script:
+  #   - $TRAVIS_BUILD_DIR/scripts/travis/deploy_server.sh

--- a/cmd/sqlflow/main.go
+++ b/cmd/sqlflow/main.go
@@ -28,7 +28,6 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
-	"syscall"
 
 	"github.com/joho/godotenv"
 	"golang.org/x/crypto/ssh/terminal"

--- a/cmd/sqlflow/main.go
+++ b/cmd/sqlflow/main.go
@@ -349,7 +349,13 @@ func main() {
 		log.Fatalf("error SQLFLOW_DATASOURCE: %v", err)
 	}
 
-	isTerminal := !flagPassed("execute", "e", "file", "f") && terminal.IsTerminal(syscall.Stdin)
+	// You might want to use syscall.Stdin instead of 0; however,
+	// unfortunately, we cannot.  the syscall standard package has
+	// a special implementation for Windows, where the type of
+	// syscall.Stdin is not int as in Linux and macOS, but
+	// uintptr.
+	isTerminal := !flagPassed("execute", "e", "file", "f") &&
+		terminal.IsTerminal(0)
 	sqlFile := os.Stdin
 
 	if flagPassed("file", "f") && *sqlFileName != "-" {

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -92,6 +92,7 @@ case "$TRAVIS_OS_NAME" in
 esac
 axel http://devtools.qiniu.com/$F.zip
 unzip $F.zip
+mv $F qshell  # Use a shorter name
 export PATH=$PWD:$PATH
 
 

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -13,7 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -x # DEBUG
 set -e
 
 # For more informaiton about deployment with Travis CI, please refer
@@ -42,7 +41,7 @@ set -e
 #     exit 1
 # fi
 
-RELEASE_TAG="latest"  # DEBUG
+RELEASE_TAG="latest"  # debug
 
 
 echo "Install download tools ..."
@@ -61,9 +60,8 @@ case "$TRAVIS_OS_NAME" in
     linux)
         # The following code snippet comes from docker/dev/install.sh
         echo "Install protoc ..."
-        PROTOC_SITE="https://github.com/protocolbuffers/protobuf/releases"
-        axel --version  # debug
-        axel $PROTOC_SITE"/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip"
+        PROTOC_SITE="https://github.com/protocolbuffers/protobuf/releases/"
+        axel --quiet $PROTOC_SITE"download/v3.7.1/protoc-3.7.1-linux-x86_64.zip"
         sudo unzip -qq protoc-3.7.1-linux-x86_64.zip -d /usr/local
         ;;
     windows) choco install protoc ;;
@@ -74,7 +72,8 @@ protoc --version
 echo "Install goyacc and protoc-gen-go ..."
 go get \
    github.com/golang/protobuf/protoc-gen-go@v1.3.3 \
-   golang.org/x/tools/cmd/goyacc
+   golang.org/x/tools/cmd/goyacc \
+   > /dev/null
 export PATH=$GOPATH/bin:$PATH
 
 
@@ -91,7 +90,7 @@ case "$TRAVIS_OS_NAME" in
     windows) F="qshell-windows-x64-v2.4.1.exe" ;;
     osx) F="qshell-darwin-x64-v2.4.1" ;;
 esac
-axel http://devtools.qiniu.com/$F.zip
+axel --quiet http://devtools.qiniu.com/$F.zip
 unzip $F.zip
 export PATH=$PWD:$PATH
 

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -62,6 +62,7 @@ case "$TRAVIS_OS_NAME" in
         # The following code snippet comes from docker/dev/install.sh
         echo "Install protoc ..."
         PROTOC_SITE="https://github.com/protocolbuffers/protobuf/releases"
+	axel --version  # debug
         axel $PROTOC_SITE"/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip"
         sudo unzip -qq protoc-3.7.1-linux-x86_64.zip -d /usr/local
         ;;

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -98,6 +98,6 @@ export PATH=$PWD:$PATH
 
 echo "Publish /tmp/sqlflow to Qiniu Object Storage ..."
 qshell account "$QINIU_AK" "$QINIU_SK" "wu"
-qshell rput sqlflow-release \
+qshell rput --overwrite sqlflow-release \
        "$RELEASE_TAG/$TRAVIS_OS_NAME/sqlflow" \
        ./sqlflow

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -81,7 +81,7 @@ export PATH=$GOPATH/bin:$PATH
 echo "Build cmd/sqlflow into /tmp ..."
 cd "$TRAVIS_BUILD_DIR"
 go generate ./...
-GOBIN=/tmp go install ./cmd/sqlflow
+GOBIN=$PWD go install ./cmd/sqlflow
 
 
 echo "Install Qiniu client for $TRAVIS_OS_NAME ..."
@@ -100,4 +100,4 @@ echo "Publish /tmp/sqlflow to Qiniu Object Storage ..."
 qshell account "$QINIU_AK" "$QINIU_SK" "wu"
 qshell rput sqlflow-release \
        "$RELEASE_TAG/$TRAVIS_OS_NAME/sqlflow" \
-       /tmp/sqlflow
+       ./sqlflow

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -44,9 +44,6 @@ set -e
 
 RELEASE_TAG="latest"  # DEBUG
 
-echo "Verify Go is installed ..."
-go env
-
 
 echo "Install download tools ..."
 case "$TRAVIS_OS_NAME" in
@@ -65,10 +62,8 @@ case "$TRAVIS_OS_NAME" in
         # The following code snippet comes from docker/dev/install.sh
         echo "Install protoc ..."
         PROTOC_SITE="https://github.com/protocolbuffers/protobuf/releases"
-        axel --quiet --output "p.zip" \
-             $PROTOC_SITE"/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip"
-        sudo unzip -qq p.zip -d /usr/local
-        rm p.zip
+        axel $PROTOC_SITE"/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip"
+        sudo unzip -qq protoc-3.7.1-linux-x86_64.zip -d /usr/local
         ;;
     windows) choco install protoc ;;
 esac
@@ -94,7 +89,7 @@ case "$TRAVIS_OS_NAME" in
     windows) F="qshell-windows-x64-v2.4.1.exe" ;;
     osx) F="qshell-darwin-x64-v2.4.1" ;;
 esac
-axel --quiet http://devtools.qiniu.com/$F.zip
+axel http://devtools.qiniu.com/$F.zip
 unzip $F.zip
 export PATH=$PWD:$PATH
 

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -19,25 +19,30 @@ set -e
 # For more informaiton about deployment with Travis CI, please refer
 # to the file header of deploy_docker.sh
 
-if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
-    echo "Skip deployment on pull request"
-    exit 0
-fi
+# DEBUG
+#
+# if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
+#     echo "Skip deployment on pull request"
+#     exit 0
+# fi
 
-# Figure out the tag to push sqlflow:ci.
-if [[ "$TRAVIS_BRANCH" == "develop" ]]; then
-    if [[ "$TRAVIS_EVENT_TYPE" == "cron" ]]; then
-        RELEASE_TAG="nightly"
-    else
-        RELEASE_TAG="latest"
-    fi
-elif [[ "$TRAVIS_TAG" != "" ]]; then
-    RELEASE_TAG="$TRAVIS_TAG"
-else
-    echo "Cannot figure out Docker image tag."
-    exit 1
-fi
+# DEBUG
+#
+# # Figure out the tag to push sqlflow:ci.
+# if [[ "$TRAVIS_BRANCH" == "develop" ]]; then
+#     if [[ "$TRAVIS_EVENT_TYPE" == "cron" ]]; then
+#         RELEASE_TAG="nightly"
+#     else
+#         RELEASE_TAG="latest"
+#     fi
+# elif [[ "$TRAVIS_TAG" != "" ]]; then
+#     RELEASE_TAG="$TRAVIS_TAG"
+# else
+#     echo "Cannot figure out Docker image tag."
+#     exit 1
+# fi
 
+RELEASE_TAG="latest"  # DEBUG
 
 echo "Verify Go is installed ..."
 go env

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -18,30 +18,25 @@ set -e
 # For more informaiton about deployment with Travis CI, please refer
 # to the file header of deploy_docker.sh
 
-# DEBUG
-#
-# if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
-#     echo "Skip deployment on pull request"
-#     exit 0
-# fi
+if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
+    echo "Skip deployment on pull request"
+    exit 0
+fi
 
-# DEBUG
-#
-# # Figure out the tag to push sqlflow:ci.
-# if [[ "$TRAVIS_BRANCH" == "develop" ]]; then
-#     if [[ "$TRAVIS_EVENT_TYPE" == "cron" ]]; then
-#         RELEASE_TAG="nightly"
-#     else
-#         RELEASE_TAG="latest"
-#     fi
-# elif [[ "$TRAVIS_TAG" != "" ]]; then
-#     RELEASE_TAG="$TRAVIS_TAG"
-# else
-#     echo "Cannot figure out Docker image tag."
-#     exit 1
-# fi
 
-RELEASE_TAG="latest"  # debug
+# Figure out the tag to push sqlflow:ci.
+if [[ "$TRAVIS_BRANCH" == "develop" ]]; then
+    if [[ "$TRAVIS_EVENT_TYPE" == "cron" ]]; then
+        RELEASE_TAG="nightly"
+    else
+        RELEASE_TAG="latest"
+    fi
+elif [[ "$TRAVIS_TAG" != "" ]]; then
+    RELEASE_TAG="$TRAVIS_TAG"
+else
+    echo "Cannot figure out Docker image tag."
+    exit 1
+fi
 
 
 echo "Install download tools ..."

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -13,6 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+set -x # DEBUG
 set -e
 
 # For more informaiton about deployment with Travis CI, please refer
@@ -42,11 +43,11 @@ echo "Verify Go is installed ..."
 go env
 
 
-echo "Install axel ..."
+echo "Install download tools ..."
 case "$TRAVIS_OS_NAME" in
     linux)
         sudo apt-get -qq update > /dev/null
-        sudo apt-get -qq install -y axel > /dev/null
+        sudo apt-get -qq install -y axel unzip > /dev/null
         ;;
     windows) choco install axel ;;
     osx) brew install axel ;;
@@ -73,11 +74,11 @@ echo "Install goyacc and protoc-gen-go ..."
 go get \
    github.com/golang/protobuf/protoc-gen-go@v1.3.3 \
    golang.org/x/tools/cmd/goyacc
-sudo cp $GOPATH/bin/* /usr/local/bin/
+export PATH=$GOPATH/bin:$PATH
 
 
 echo "Build cmd/sqlflow into /tmp ..."
-cd $TRAVIS_BUILD_DIR
+cd "$TRAVIS_BUILD_DIR"
 go generate ./...
 GOBIN=/tmp go install ./cmd/sqlflow
 
@@ -90,11 +91,11 @@ case "$TRAVIS_OS_NAME" in
 esac
 axel --quiet http://devtools.qiniu.com/$F.zip
 unzip $F.zip
-sudo mv $F /usr/local/bin/qshell
+export PATH=$PWD:$PATH
 
 
 echo "Publish /tmp/sqlflow to Qiniu Object Storage ..."
 qshell account "$QINIU_AK" "$QINIU_SK" "wu"
 qshell rput sqlflow-release \
-       $RELEASE_TAG/$TRAVIS_OS_NAME/sqlflow \
+       "$RELEASE_TAG/$TRAVIS_OS_NAME/sqlflow" \
        /tmp/sqlflow

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -97,6 +97,7 @@ export PATH=$PWD:$PATH
 
 echo "Publish /tmp/sqlflow to Qiniu Object Storage ..."
 $F account "$QINIU_AK" "$QINIU_SK" "wu"
-$F rput --overwrite sqlflow-release \
-       "$RELEASE_TAG/$TRAVIS_OS_NAME/sqlflow" \
-       ./sqlflow
+$F rput --overwrite \
+   sqlflow-release \
+   "$RELEASE_TAG/$TRAVIS_OS_NAME/sqlflow" \
+   ./sqlflow

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -92,12 +92,11 @@ case "$TRAVIS_OS_NAME" in
 esac
 axel http://devtools.qiniu.com/$F.zip
 unzip $F.zip
-mv $F qshell  # Use a shorter name
 export PATH=$PWD:$PATH
 
 
 echo "Publish /tmp/sqlflow to Qiniu Object Storage ..."
-qshell account "$QINIU_AK" "$QINIU_SK" "wu"
-qshell rput --overwrite sqlflow-release \
+$F account "$QINIU_AK" "$QINIU_SK" "wu"
+$F rput --overwrite sqlflow-release \
        "$RELEASE_TAG/$TRAVIS_OS_NAME/sqlflow" \
        ./sqlflow

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -80,8 +80,9 @@ export PATH=$GOPATH/bin:$PATH
 
 echo "Build cmd/sqlflow into /tmp ..."
 cd "$TRAVIS_BUILD_DIR"
-go generate ./...
-GOBIN=$PWD go install ./cmd/sqlflow
+go generate ./... > /dev/null
+mkdir "$PWD"/build
+GOBIN="$PWD"/build go install ./cmd/sqlflow > /dev/null
 
 
 echo "Install Qiniu client for $TRAVIS_OS_NAME ..."
@@ -100,4 +101,4 @@ $F account "$QINIU_AK" "$QINIU_SK" "wu"
 $F rput --overwrite \
    sqlflow-release \
    "$RELEASE_TAG/$TRAVIS_OS_NAME/sqlflow" \
-   ./sqlflow
+   "$PWD"/build/sqlflow*  # Need * because for Windows it is sqlflow.exe

--- a/scripts/travis/deploy_client.sh
+++ b/scripts/travis/deploy_client.sh
@@ -52,7 +52,7 @@ case "$TRAVIS_OS_NAME" in
         sudo apt-get -qq install -y axel unzip > /dev/null
         ;;
     windows) choco install axel ;;
-    osx) brew install axel ;;
+    osx) brew install axel > /dev/null ;;
 esac
 
 
@@ -62,7 +62,7 @@ case "$TRAVIS_OS_NAME" in
         # The following code snippet comes from docker/dev/install.sh
         echo "Install protoc ..."
         PROTOC_SITE="https://github.com/protocolbuffers/protobuf/releases"
-	axel --version  # debug
+        axel --version  # debug
         axel $PROTOC_SITE"/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip"
         sudo unzip -qq protoc-3.7.1-linux-x86_64.zip -d /usr/local
         ;;

